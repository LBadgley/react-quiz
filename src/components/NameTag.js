import React, { PureComponent } from 'react';
import ColorSelector from './ColorSelector';
import Shape from './Shape';
import TextSelector from './TextSelector';

export default class NameTag extends PureComponent {
  state = {
    text: '',
    color: '',
  }
  changeHandler = ({ target }) => {
    this.setState({ [target.name]: target.value });
  }

  render() {
    const { text, color } = this.state;

    return (
      <section className={NameTag}>
        <input type="text" name="text" value={text} onChange={this.changeHandler} />  
        <input name="color" type="color" value={color} onChange={this.changeHandler} />
        <input name="color" type="color" value={color} onChange={this.changeHandler} />
        <div className={[color]}>{text}</div>

      </section>
    );
  }
}
