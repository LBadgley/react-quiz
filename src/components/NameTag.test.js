import React from 'react';
import { shallow } from 'enzyme';
import NameTag from './NameTag';

describe('nameTag component', () => {
  it('renders the name tag', () => {
    const wrapper = shallow(<NameTag />);
    expect(wrapper).toMatchSnapshot();
  });
});
