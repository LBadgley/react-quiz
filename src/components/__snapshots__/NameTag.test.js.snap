// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nameTag component renders the name tag 1`] = `
<section>
  <input
    name="text"
    onChange={[Function]}
    type="text"
    value=""
  />
  <input
    name="color"
    onChange={[Function]}
    type="color"
    value=""
  />
  <div
    className={
      Array [
        [Function],
      ]
    }
  />
</section>
`;
